data-access-layer/cf/UaaClient.js

```diff
@@ -22,11 +22,13 @@ class UaaClient extends HttpClient {
     this.clientId = config.cf.client_id || 'cf';
   }
 
-  authorizationUrl(options) {
+  authorizationUrl(options, addLoginHint) {
```

Comment:
here why is this a boolean ? Ideally it will have a value as "uaa" right ? So if its a non null string we can directly assign it to the origin value below .. So that it becomes extensible in future.

```
if (addLoginHint!==null && addLoginHint !== undefined) {
      options.login_hint = '{"origin":addLoginHint}';
    }
```

something like this.

Reply:
My understanding was that login_hint will be needed for UAA case only, hence the hardcoding here. However, in the view of extensibility will make this change.

Reply:
Yes , that is correct . But in future if origin is different then we would not need changes in the broker.

Reply:
Understood. Will push the change shortly